### PR TITLE
Add OpInfo for torch.linalg.tensorsolve

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds an OpInfo entry for tensorsolve function.
The keyword argument is different from NumPy so a lambda function is needed to be passed to `ref=`.
I had to change the dtypes for `test_reference_testing` because NumPy does computation internally using double for all linear algebra functions and maybe for some other functions. Using `torch.float64` and `torch.complex128` is more reliable for NumPy comparisons.

cc @mruberry

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def test_multiple_devices(self, devices, dtype, op):
     #   values on the tensors from sample_inputs func for the corresponding op.
     @onlyNativeDeviceTypes
     @suppress_warnings
-    @ops(_ref_test_ops, allowed_dtypes=(torch.float32, torch.long, torch.complex64))
+    @ops(_ref_test_ops, allowed_dtypes=(torch.float64, torch.long, torch.complex128))
```
Comment (mruberry) [inline test/test_ops.py:203]: This change makes sense give how often NumPy promotes to double, but it's also a big change.

How much longer does this test take to run than when it uses float32 and complex64?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -6962,6 +6962,18 @@ def make_input():
 
     return samples
 
+def sample_inputs_tensorsolve(op_info, device, dtype, requires_grad, **kwargs):
+    a_shapes = [(2, 3, 6), (3, 4, 4, 3)]
+    dimss = [None, (0, 2)]
+
+    def gen_inputs():
+        for a_shape, dims in itertools.product(a_shapes, dimss):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:6970]: This is fine but just a few handwritten inputs that test interesting cases would be a great alternative

File: test/test_ops.py (hunk 1 above)
Comment (IvanYashchuk) [reply test/test_ops.py:203]: It's actually faster on my computer with the proposed change. It took 63.45 seconds with float32 and complex64 and 62.10 seconds with float64 and complex128.

```
python -m pytest test/test_ops.py -k 'test_reference_testing'

# float32 and complex64
1 failed, 180 passed, 91 skipped, 32563 deselected, 2 warnings in 63.45s (0:01:03)

# float64 and complex128:
181 passed, 91 skipped, 32563 deselected, 2 warnings in 62.10s (0:01:02)
```

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (IvanYashchuk) [reply torch/testing/_internal/common_methods_invocations.py:6970]: I wanted to add also shapes with 0, but NumPy fails then. It's difficult to be creative here: this function reshapes the input to 2D and calls `torch.linalg.solve` internally; besides 0 in the shape nothing else seems interesting.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:6970]: Adding zero would be interesting. I'd at least add the case and comment it out with a note about the NumPy failure. Do you want to report a bug to NumPy?

File: test/test_ops.py (hunk 1 above)
Comment (mruberry) [reply test/test_ops.py:203]: Now how the heck is that even possible.

But OK I suppose my concern is addressed.

**Review by mruberry — APPROVED** [review]
One inline comment for your review, @IvanYashchuk 

Would you also add a comment before the reference test about why it runs in double and complex double? Then just ping me when you'd like this merged

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (IvanYashchuk) [reply torch/testing/_internal/common_methods_invocations.py:7033]: I added a zero dim case and submitted a bug fix to NumPy. With the bug fix, all tests pass for sample input with zero dimensions.

**Comment by IvanYashchuk** [comment]
@mruberry, I added a comment about the choice of double precision for reference tests.
This PR is ready to be merged.
